### PR TITLE
Add: Setting to toggle chemical side mission and chemical weapon cache probability

### PR DESCRIPTION
**Pull request by TomDraal**
Hi,
I've made some changes in order to add some more advanced settings on the Chemical Warfare in H&M.
The idea is to enable chemical warfare and choose if the chemical side missions will spawn or not and also set a chance probability of having a chemical weapons cache or not.

- Add: Setting to toggle chemical side mission (@TomDraal).
- Add: Setting to change chemical weapon cache probability (@TomDraal).

**When merged this pull request will:**
- Allow the user to choose if he/she wants to have chemical side missions while chemical warfare is enabled
- Tweak the chance probability (%age) of having a chemical weapons cache

**Final test:**
- [X] local
- [X] server

### CODE REVIEW

**Comment by Vdauphin** [comment]
Hello,

Thank you for your contribution
We need to be consistent between setting
Before: `btc_p_chem` ON, side mission and chemical were enabled
Now, if `btc_p_chem` ON but `btc_p_chem_sides` and `btc_p_chem_cache_probability` are OFF, nothing will happen

Despite, I don't really see the point of having a setting for side mission as they can be skipped by the Officer I would recommend to 
- add the existing setting `btc_p_chem`: Off, On without weapon cache, On without side mission, On with both
- create a `btc_p_chem_cache_probability` as you did without the 0%

Cheers

**Comment by TomDraal** [comment]
Hello,
We want to be able to deactivate the chemical side missions because sometimes while using the skip feature some contaminated elements remains. At least we had some bad surprises in the past, so deactivating the chemical side missions is the safest way.
If you want full consistency then the best way is to completely remove `btc_p_chem` which I saw like a general switch on / off.
So I wanted to keep it just in case someone would want to add other chemical warfare related tasks.
By only keeping `btc_p_chem_sides` and `btc_p_chem_cache_probability` with a 0-100% range, we will have the behaviour you described : 
- sides ON and cache OFF (0%)
- sides OFF and cache ON (>0%)
- sides OFF and cache OFF (0%) => equivalent to btc_p_chem OFF
- sides ON and cache ON (>0%) => equivalent to btc_p_chem ON
Would that suits you ?
Cheers

**Comment by Vdauphin** [comment]
> If you want full consistency then the best way is to completely remove btc_p_chem which I saw like a general switch on / off.

I thought about it but I think it would be simpler to use number instead of bool for `btc_p_chem` (just do  `btc_p_chem > 0` where needed)
It is up to you to make a choice on this!

File: =BTC=co@30_Hearts_and_Minds.Altis/core/fnc/cache/create.sqf [hunk 1]
```diff
@@ -28,13 +28,13 @@ Author:
 
 params [
     ["_cache_pos", btc_cache_pos, [[]]],
-    ["_p_chem", btc_p_chem, [true]],
-    ["_probabilityNotChemical", 0.5, [0]]
+    ["_p_chem", (btc_p_chem_cache_probability > 0), [true]],
```
Comment (Vdauphin) [inline =BTC=co@30_Hearts_and_Minds.Altis/core/fnc/cache/create.sqf:31]:
```suggestion
    ["_p_chem", btc_p_chem_cache_probability > 0, [true]],
```

File: =BTC=co@30_Hearts_and_Minds.Altis/core/def/param.hpp [hunk 2]
```diff
@@ -401,12 +401,18 @@ class Params {
         texts[] = {$STR_DISABLED,$STR_ENABLED};
         default = 1;
     };
-    class btc_p_chem { // Chemical warfare
-        title = __EVAL(format ["      %1", localize "STR_BTC_HAM_PARAM_GAMEPLAY_CHEM"]);
+    class btc_p_chem_sides { // Advanced Chemical warfare setting to activate / deactivate chemical side missions
```
Comment (Vdauphin) [inline =BTC=co@30_Hearts_and_Minds.Altis/core/def/param.hpp:404]:
```suggestion
    class btc_p_chem_sides { // Toggle chemical side missions:
```

File: =BTC=co@30_Hearts_and_Minds.Altis/core/def/param.hpp [hunk 3]
```diff
@@ -401,12 +401,18 @@ class Params {
...
+        title = __EVAL(format ["      %1", localize "STR_BTC_HAM_PARAM_GAMEPLAY_CHEM_SIDES"]);
         values[] = {0,1};
         texts[] = {$STR_DISABLED,$STR_ENABLED};
         default = 1;
     };
+    class btc_p_chem_cache_probability { // Advanced Chemical warfare setting to change probability of a chemical weapon cache
```
Comment (Vdauphin) [inline =BTC=co@30_Hearts_and_Minds.Altis/core/def/param.hpp:410]:
```suggestion
    class btc_p_chem_cache_probability { // Chemical weapon cache probability:
```

File: =BTC=co@30_Hearts_and_Minds.Altis/core/fnc/chem/checkLoop.sqf [hunk 4]
```diff
@@ -19,7 +19,9 @@ Author:
 
 ---------------------------------------------------------------------------- */
 
-if !(btc_p_chem) exitWith {};
+private _isChem = (btc_p_chem_sides || (btc_p_chem_cache_probability > 0));
```
Comment (Vdauphin) [inline =BTC=co@30_Hearts_and_Minds.Altis/core/fnc/chem/checkLoop.sqf:22]: One line:

```sqf
if !(btc_p_chem_sides || (btc_p_chem_cache_probability > 0)) exitWith {};
```

File: =BTC=co@30_Hearts_and_Minds.Altis/core/fnc/chem/handleShower.sqf [hunk 5]
```diff
@@ -20,7 +20,9 @@ Author:
 
 ---------------------------------------------------------------------------- */
 
-if !(btc_p_chem) exitWith {};
+private _isChem = (btc_p_chem_sides || (btc_p_chem_cache_probability > 0));
```
Comment (Vdauphin) [inline =BTC=co@30_Hearts_and_Minds.Altis/core/fnc/chem/handleShower.sqf:23]: One line and no extra parenthesis

File: =BTC=co@30_Hearts_and_Minds.Altis/core/fnc/db/load.sqf [hunk 6]
```diff
@@ -102,7 +102,7 @@ btc_cache_pos = _cache_pos;
 btc_cache_n = _cache_n;
 btc_cache_info = _cache_info;
 
-[_cache_pos, btc_p_chem, [1, 0] select _isChem] call btc_cache_fnc_create;
+[_cache_pos, (btc_p_chem_cache_probability > 0), [1, 0] select _isChem] call btc_cache_fnc_create;
```
Comment (Vdauphin) [inline =BTC=co@30_Hearts_and_Minds.Altis/core/fnc/db/load.sqf:105]:
```suggestion
[_cache_pos, btc_p_chem_cache_probability > 0, [1, 0] select _isChem] call btc_cache_fnc_create;
```

File: =BTC=co@30_Hearts_and_Minds.Altis/core/fnc/db/load_old.sqf [hunk 7]
```diff
@@ -100,7 +100,7 @@ btc_cache_pos = _cache_pos;
 btc_cache_n = _cache_n;
 btc_cache_info = _cache_info;
 
-[_cache_pos, btc_p_chem, [1, 0] select _isChem] call btc_cache_fnc_create;
+[_cache_pos, (btc_p_chem_cache_probability > 0), [1, 0] select _isChem] call btc_cache_fnc_create;
```
Comment (Vdauphin) [inline =BTC=co@30_Hearts_and_Minds.Altis/core/fnc/db/load_old.sqf:103]:
```suggestion
[_cache_pos, btc_p_chem_cache_probability > 0, [1, 0] select _isChem] call btc_cache_fnc_create;
```

File: =BTC=co@30_Hearts_and_Minds.Altis/core/fnc/eh/player.sqf [hunk 8]
```diff
@@ -71,7 +71,9 @@ _player addEventHandler ["WeaponAssembled", {
     _this remoteExecCall ["btc_log_fnc_init", 2];
 }] call CBA_fnc_addEventHandler;
 
-if (btc_p_chem) then {
+private _isChem = (btc_p_chem_sides || (btc_p_chem_cache_probability > 0));
```
Comment (Vdauphin) [inline =BTC=co@30_Hearts_and_Minds.Altis/core/fnc/eh/player.sqf:74]: one line

File: =BTC=co@30_Hearts_and_Minds.Altis/core/fnc/eh/server.sqf [hunk 9]
```diff
@@ -85,7 +85,10 @@ if (btc_p_auto_db) then {
         };
     }];
 };
-if (btc_p_chem) then {
+
+private _isChem = (btc_p_chem_sides || (btc_p_chem_cache_probability > 0));
```
Comment (Vdauphin) [inline =BTC=co@30_Hearts_and_Minds.Altis/core/fnc/eh/server.sqf:89]: one line

**Review by Vdauphin — CHANGES_REQUESTED** [review]
thanks for the update, some suggestion her

File: =BTC=co@30_Hearts_and_Minds.Altis/stringtable.xml [hunk 10]
```diff
@@ -961,6 +961,14 @@
                 <French>Guerre chimique:</French>
                 <Czech>Chemická válka:</Czech>
             </Key>
+            <Key ID="STR_BTC_HAM_PARAM_GAMEPLAY_CHEM_SIDES">
+                <Original>Chemical side missions:</Original>
+                <French>Missions secondaires chimiques:</French>
+            </Key>
+            <Key ID="STR_BTC_HAM_PARAM_GAMEPLAY_CHEM_CACHE_PROBABILITY">
+                <Original>Chemical weapon cache probability:</Original>
+                <French>Probabilité de cache d'armes chimique:</French>
```
Comment (Vdauphin) [inline =BTC=co@30_Hearts_and_Minds.Altis/stringtable.xml:970]:
```suggestion
                <French>Probabilité de cache d'armes chimiques :</French>
```

File: =BTC=co@30_Hearts_and_Minds.Altis/stringtable.xml [hunk 11]
```diff
@@ -961,6 +961,14 @@
                 <French>Guerre chimique:</French>
                 <Czech>Chemická válka:</Czech>
             </Key>
+            <Key ID="STR_BTC_HAM_PARAM_GAMEPLAY_CHEM_SIDES">
+                <Original>Chemical side missions:</Original>
+                <French>Missions secondaires chimiques:</French>
```
Comment (Vdauphin) [inline =BTC=co@30_Hearts_and_Minds.Altis/stringtable.xml:966]:
```suggestion
                <French>Missions secondaires chimiques :</French>
```

**Review by Vdauphin — CHANGES_REQUESTED** [review]
Some tiny change!

**Review by Vdauphin — APPROVED** [review]

**Pull request merged**